examples/run_from_pickle.py

```diff
@@ -0,0 +1,31 @@
+# run from a simulation state pickle to help with debugging
```

Comment:
Add a test for this in `tests/test_examples.py`?

Reply:
i didn't mean to check this in - the tests already test reproduciblity from the `(initial_state, protocol)` pair